plugin/src/main/java/com/denizenscript/denizen/objects/NPCTag.java

```diff
@@ -1307,6 +1369,101 @@ else if (attribute.startsWith("list", 2)) {
             }
         });
+
+        // <--[mechanism]
+        // @object NPCTag
+        // @name waypoint_provider
+        // @input ElementTag
+        // @description
+        // Sets the NPC's Waypoints Provider (default options: linear, wander, guided.)
+        // @tags
+        // <NPCTag.waypoint_provider>
+        // -->
+        tagProcessor.registerMechanism("waypoint_provider", false, ElementTag.class, (object, mechanism, input) -> {
+            Waypoints wp = object.getCitizen().getOrAddTrait(Waypoints.class);
+            wp.setWaypointProvider(input.asString());
+        });
+
+        // <--[mechanism]
+        // @object DurationTag
+        // @name wander_delay
+        // @input DurationTag
+        // @description
+        // Sets the delay (in seconds) for an NPC's wander Waypoints Provider. Provider must be set to Wander before setting wander_delay. Set to -1 to use Mincraft's default value.
+        // @tags
+        // <NPCTag.wander_delay>
+        // -->
+        tagProcessor.registerMechanism("wander_delay", false, DurationTag.class, (object, mechanism, input) -> {
+            Waypoints wp = object.getCitizen().getOrAddTrait(Waypoints.class);
+            if (wp.getCurrentProvider() instanceof WanderWaypointProvider wanderWaypointProvider) {
+                if (input.getSecondsAsInt() >= -1) {
+                    wanderWaypointProvider.setDelay(input.getSecondsAsInt());
+                }
+                else {
+                    mechanism.echoError("Invalid value for wander delay");
+                }
+            }
+        });
+
+        // <--[mechanism]
+        // @object NPCTag
+        // @name wander_worldguardregion
+        // @input ElementTag
+        // @description
+        // Sets the worldguardRegion for an NPC's wander Waypoints Provider.  
+        // @tags
+        // <NPCTag.wander_worldguardregion>
+        // -->
+        tagProcessor.registerMechanism("wander_worldguardregion", false, ElementTag.class, (object, mechanism, input) -> {
+            Waypoints wp = object.getCitizen().getOrAddTrait(Waypoints.class);
+            if (wp.getCurrentProvider() instanceof WanderWaypointProvider wanderWaypointProvider) {
+                    wanderWaypointProvider.setWorldGuardRegion(input.asString());
+            }
+        });
+
+        // <--[mechanism]
+        // @object NPCTag
+        // @name wander_xrange
+        // @input ElementTag(Number)
+        // @description
+        // Sets the xrange for an NPC's wander path.  
+        // @tags
+        // <NPCTag.wander_xrange>
+        // -->
+        tagProcessor.registerMechanism("wander_xrange", false, ElementTag.class, (object, mechanism, input) -> {
+            Waypoints wp = object.getCitizen().getOrAddTrait(Waypoints.class);
+            if (wp.getCurrentProvider() instanceof WanderWaypointProvider wanderWaypointProvider) {
+                int yrange = wanderWaypointProvider.getYRange();
```

Comment:
wrong var name (copypasta)

Reply:
The method in Citizens is for setting an XY range, and at one point someone indicated it would be better to have one mechanism for each X and Y, so when one is set, it queries the other and passes the pair to Citizens.

Reply:
Oh, you're right, my bad